paddle/fluid/framework/details/reduce_op_handle.cc

```diff
@@ -91,11 +91,26 @@ void ReduceOpHandle::RunImpl() {
   } else {
     std::vector<const LoDTensor *> lod_tensors =
         GetInputValues<LoDTensor>(in_var_handles, var_scopes);
+
     if (paddle::platform::is_cpu_place(lod_tensors[0]->place())) {
       this->RunAndRecordEvent([&] {
-        ReduceLoDTensor func(lod_tensors,
-                             out_var->GetMutable<framework::LoDTensor>());
+        // FIXME(zcd): The order of summing is important,
+        // especially when the type of data is float or double.
+        // For example, the result of `a+b+c+d` may be different
+        // with the result of `c+a+b+d`, so the summing order should be fixed.
```

Comment:
need a little detail here, the reduce implementation is at `reduce_and_gather.h`, so people would not quite understand why using scope 0 tensor could do the ordering.

Reply:
And do we need some flag here, for keeping the order seems will make it slow.

Reply:
Great! I have added a flag, `cpu_deterministic`